### PR TITLE
[extractor/wordpress:mb.miniAudioPlayer] Add generic embed extractor

### PR DESCRIPTION
**Pull request by coletdjnz**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

ADD DESCRIPTION HERE

Fixes https://github.com/yt-dlp/yt-dlp/issues/4994

Adds a generic extractor for the wordpress plugin version of mb.miniAudioPlayer (https://wordpress.org/plugins/wp-miniaudioplayer/)

This plugin uses jquery to select anchor elements on the page with href pointing to an audio file. This is more or less hard-coded across versions.

I have not found any cases for tests making use of the css excludes functionality (i.e., have elements with the css class that would need to be excluded). I have added some basic filtering but cannot be sure that is how it works in practice.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [X] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/wordpress.py [hunk 1]
```diff
@@ -67,3 +72,87 @@ def _extract_from_webpage(self, url, webpage):
...
+    }, {
+        # Version 1.9.7: https://plugins.trac.wordpress.org/browser/wp-miniaudioplayer/tags/1.9.7
+        # But has spaces around href filter
+        'url': 'https://www.estudiords.com.br/temas/',
+        'info_dict': {
+            'id': 'temas',
+            'title': 'Temas Variados',
+            'age_limit': 0,
+            'timestamp': float,
+            'upload_date': str,
```
Comment (pukkandan) [inline yt_dlp/extractor/wordpress.py:122]: do these change?

File: yt_dlp/extractor/wordpress.py [hunk 2]
```diff
@@ -67,3 +72,87 @@ def _extract_from_webpage(self, url, webpage):
...
+        # v1.9.4+ has "a[href*='.mp3']" only
+        file_exts = re.findall(r'a\[href\s*\*=\s*\'\.([a-zA-Z\d]+)\'', mb_player_params)
+        if not file_exts:
+            return
+
+        # XXX: this is inefficient, but it works
+        candidates = (
+            get_element_text_and_html_by_tag('a', html)
+            for _, html in get_elements_text_and_html_by_attribute(
+                'href', rf'(?:[^\"\']+\.(?:{"|".join(file_exts)}))', webpage, escape_value=False))
```
Comment (pukkandan) [inline yt_dlp/extractor/wordpress.py:146]: We could add an optional `tag` parameter to `get_elements_text_and_html_by_attribute`

File: yt_dlp/extractor/wordpress.py (hunk 1 above)
Comment (coletdjnz) [reply yt_dlp/extractor/wordpress.py:122]: yeah, it's from generic extracting pulling the modified date header I think

Comment (pukkandan) [reply yt_dlp/extractor/wordpress.py:122]: If it is wrong, you can override it by returning `None`

File: yt_dlp/extractor/wordpress.py (hunk 2 above)
Comment (coletdjnz) [reply yt_dlp/extractor/wordpress.py:146]: hmm. Could do something like

```diff
diff --git a/yt_dlp/utils.py b/yt_dlp/utils.py
index 3e2ce8434..746ac0295 100644
--- a/yt_dlp/utils.py
+++ b/yt_dlp/utils.py
@@ -408,7 +408,7 @@ def get_elements_html_by_attribute(*args, **kwargs):
     return [whole for _, whole in get_elements_text_and_html_by_attribute(*args, **kwargs)]
 
 
-def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value=True):
+def get_elements_text_and_html_by_attribute(attribute, value, html, tag=r'[a-zA-Z0-9:._-]+', escape_value=True):
     """
     Return the text (content) and the html (whole) of the tag with the specified
     attribute in the passed HTML document
@@ -419,7 +419,7 @@ def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value
     value = re.escape(value) if escape_value else value
 
     partial_element_re = rf'''(?x)
-        <(?P<tag>[a-zA-Z0-9:._-]+)
+        <(?P<tag>{tag})
          (?:\s(?:[^>"']|"[^"]*"|'[^']*')*)?
          \s{re.escape(attribute)}\s*=\s*(?P<_q>['"]{quote})(?-x:{value})(?P=_q)
         '''

```

Comment (pukkandan) [reply yt_dlp/extractor/wordpress.py:146]: looks fine

File: yt_dlp/utils.py [hunk 3]
```diff
@@ -408,7 +408,7 @@ def get_elements_html_by_attribute(*args, **kwargs):
     return [whole for _, whole in get_elements_text_and_html_by_attribute(*args, **kwargs)]
 
 
-def get_elements_text_and_html_by_attribute(attribute, value, html, escape_value=True):
+def get_elements_text_and_html_by_attribute(attribute, value, html, tag=r'[a-zA-Z0-9:._-]+', escape_value=True):
```
[the remark this answers is not in the document]
Comment (coletdjnz) [reply yt_dlp/utils.py:411]: unlike the value, tag should not need to be escaped imo.

**Pull request merged**